### ISSUE TITLE
Location input is not updating aria-expanded correctly

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22886]
## Questions for refinement
- 

---
## Describe the defect
**Note: this ticket should be completed before #22830** because it will make it easier to implement and test if this is done first.

aria-expanded notifies assitive tech when a dropdown is expanded or collapsed. This usually triggers a screenreader announcement tolet screenreader users know when a dropdown is open or closed. The location field dropdown is occasionally correct in matching the aria-expanded state to the actual state of the dropdown, but sometimes there’s a mismatch. Sometimes the dropdown is actually open when aria-expanded=false which indicates to screenreader users that it is closed. The opposite is also true. This mismatch means that screenreader users cannot reliably know when the dropdown is truly expanded or collapsed.

 - **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 1
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User can perceive all meaningful elements, User can identify an element and its state

## To Reproduce
1. Go to Facility Locator
2. Open devtools.
3. Open and close the dropdown. Try tabbing to the field, clicking, typing.
4. Note that the aria-expanded value does not always match the actual state of the dropdown.
5. Also try a screenreader while opening and closing the dropdown.
6. Note that sometimes the dropdown is announced as expanded when it's not. Or announces it's collapsed when it's open.

## Screenshots
Location input isn't showing a dropdown, but aria-expanded is set to true. 😳

<img width="1742" height="705" alt="Image" src="https://github.com/user-attachments/assets/0f04eab0-7eb4-4529-968e-b775e1a54c43" />

## Recommendation
Ensure aria-expanded is true only when the dropdown is expanded and is false when dropdown is collapsed.

## Additional context
- 

## AC / Expected behavior
For the location input,
- [ ] aria-expanded=true when the dropdown is open
- [ ] aria-expanded=false when the dropdown is closed

## Definition of done
- [x] Code completed; PR created
- [x] Review instance created/available; internal team notified to start reviews
- [ ] Code review (meets established quality and security standards)
- [ ] ~Design review (meets design & UX requirements)~
- [x] A11y review (meets accessibility standards) REQUIRED
- [x] Functional/feature review (meets acceptance criteria)
- [ ] Automated tests passed; PR approved
- [x] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by r-bartlett-gsa** [comment]
@qjac This is ready for a11y testing: http://ec0633ca8ea69ef3504f2645335b5caa.vfs.va.gov/ Thank you!!

**Comment by r-bartlett-gsa** [comment]
@carlyMhayter This is good to go as soon as the PR is reviewed and approved.
